app/models/security_activity.rb

```diff
@@ -1,52 +1,68 @@
 class SecurityActivity < ApplicationRecord
-  enum event_type: {
-    login: 0,
-    change_email: 1,
-    change_phone: 2,
-    change_password: 3, # pragma: allowlist secret
-  }
-
-  belongs_to :user
-
-  def self.login!(user, ip_address)
-    new(
-      event_type: :login,
-      user_id: user.id,
-      ip_address: ip_address,
-    ).save!
-  end
+  # TODO: decide which of these to show on the security page
+  EVENTS = [
+    # logging in
+    ACCOUNT_LOCKED = LogEntry.new(id: 5, name: :account_locked, require_user: true),
+    MANUAL_ACCOUNT_UNLOCK = LogEntry.new(id: 6, name: :manual_account_unlock, require_user: true),
 
-  def self.login_with!(user, oauth_application, ip_address)
-    new(
-      event_type: :login,
-      user_id: user.id,
-      oauth_application_id: oauth_application.id,
-      ip_address: ip_address,
-    ).save!
-  end
+    TWO_STEP_VERIFICATION_SUCCESS = LogEntry.new(id: 7, name: :two_step_verification_success, require_user: true),
+    TWO_STEP_VERIFICATION_FAILURE = LogEntry.new(id: 8, name: :two_step_verification_failure, require_user: true),
+
+    LOGIN_SUCCESS = LogEntry.new(id: 0, name: :login_success, require_user: true),
+    LOGIN_FAILURE = LogEntry.new(id: 9, name: :login_failure, require_user: true),
+
+    # recovery
+    PASSWORD_RESET_REQUEST = LogEntry.new(id: 10, name: :password_reset_request, require_user: true),
+    PASSWORD_RESET_SUCCESS = LogEntry.new(id: 4, name: :password_reset_success, require_user: true),
+
+    # update
+    EMAIL_CHANGED = LogEntry.new(id: 1, name: :email_changed, require_user: true),
+    TWO_STEP_CHANGED = LogEntry.new(id: 2, name: :two_step_changed, require_user: true),
+    PASSWORD_CHANGED = LogEntry.new(id: 3, name: :password_changed, require_user: true),
+
+    EMAIL_CHANGE_CONFIRMED = LogEntry.new(id: 11, name: :email_change_confirmed, require_user: true),
```

Comment:
Could this be moved up a few lines to be next to the `EMAIL_CHANGED` value, to keep the email change stuff together